eth/tracers/api_blocktrace.go

```diff
@@ -299,22 +315,47 @@ func (api *API) getTxResult(env *traceEnv, state *state.StateDB, index int, bloc
 
 	proofStorages := tracer.UpdatedStorages()
 	for addr, keys := range proofStorages {
-		for key := range keys {
+		env.sMu.Lock()
+		trie, err := state.GetStorageTrieForProof(addr)
+		if err != nil {
+			// but we still continue to next address
```

Comment:
Will this lead to corrupted traces?

Reply:
If this had happened, it should be caused **by** correupted trace (so the tracer have caught some stores but the account uder these ops is not existed in trie in fact).

Figure out that this may be happend while trying some SSTORE/SLOAD on an unexisted account. I would update here for always returning a trie (even just an empty one) so we can always obtain some proof data.